src/app/clusters/service-area-server/service-area-server.cpp

```diff
@@ -0,0 +1,1101 @@
+
+    // Check if the cluster has been selected in zap
+    VerifyOrReturnError(emberAfContainsServer(mEndpointId, chip::app::Clusters::ServiceArea::Id), CHIP_ERROR_INVALID_ARGUMENT,
+                        ChipLogError(Zcl, "Service Area: The cluster with Id %lu was not enabled in zap.",
+                                     long(chip::app::Clusters::ServiceArea::Id)));
+
+    ReturnErrorOnFailure(InteractionModelEngine::GetInstance()->RegisterCommandHandler(this));
+
+    VerifyOrReturnError(registerAttributeAccessOverride(this), CHIP_ERROR_INCORRECT_STATE);
+
+    ReturnErrorOnFailure(mDelegate->Init());
+
+    return CHIP_NO_ERROR;
+}
+
+//*************************************************************************
+// core functions
+
+CHIP_ERROR Instance::Read(const ConcreteReadAttributePath & aPath, AttributeValueEncoder & aEncoder)
+{
+    ChipLogError(Zcl, "Service Area: Reading");
+
+    switch (aPath.mAttributeId)
+    {
+
+    case ServiceArea::Attributes::SupportedLocations::Id: {
+        ChipLogProgress(Zcl, "Service Area: Read SupportedLocations");
+        ReturnErrorOnFailure(ReadSupportedLocations(aEncoder));
+        break;
+    }
+
+    case ServiceArea::Attributes::SupportedMaps::Id: {
+        ChipLogProgress(Zcl, "Service Area: Read SupportedMaps");
+        ReturnErrorOnFailure(ReadSupportedMaps(aEncoder));
+        break;
+    }
+
+    case ServiceArea::Attributes::SelectedLocations::Id: {
+        ChipLogProgress(Zcl, "Service Area: Read SelectedLocations");
+        ReturnErrorOnFailure(ReadSelectedLocations(aEncoder));
+        break;
+    }
+
+    case ServiceArea::Attributes::CurrentLocation::Id: {
+        ChipLogProgress(Zcl, "Service Area: Read CurrentLocation");
+        ReturnErrorOnFailure(aEncoder.Encode(GetCurrentLocation()));
+        break;
+    }
+
+    case ServiceArea::Attributes::EstimatedEndTime::Id: {
+        ChipLogProgress(Zcl, "Service Area: Read EstimatedEndTime");
+        ReturnErrorOnFailure(aEncoder.Encode(GetEstimatedEndTime()));
+        break;
+    }
+
+    case ServiceArea::Attributes::Progress::Id: {
+        ChipLogProgress(Zcl, "Service Area: Read Progress");
+        ReturnErrorOnFailure(ReadProgress(aEncoder));
+        break;
+    }
+
+    case ServiceArea::Attributes::FeatureMap::Id: {
+        ReturnErrorOnFailure(aEncoder.Encode(mFeature.Raw()));
+        break;
+    }
+
+    default:
+        ChipLogProgress(Zcl, "Service Area: Read unsupported attribute %u", aPath.mAttributeId);
+        break;
+    }
+
+    return CHIP_NO_ERROR;
+}
+
+void Instance::InvokeCommand(HandlerContext & handlerContext)
+{
+    ChipLogProgress(Zcl, "%s", __func__);
+    switch (handlerContext.mRequestPath.mCommandId)
+    {
+    case Commands::SelectLocations::Id:
+
+        CommandHandlerInterface::HandleCommand<Commands::SelectLocations::DecodableType>(
+            handlerContext, [this](HandlerContext & ctx, const auto & req) { HandleSelectLocationsCmd(ctx, req); });
+        break;
```

Comment:
could we just `return` instead of the logic of `break and then return`? Reduces the scope of following code paths.

Reply:
Addressed [here](https://github.com/project-chip/connectedhomeip/pull/33991/commits/2b9f11210c6d36f8088fe2a9c86ba4a2a1065029).